lightning/src/onion_message/messenger.rs

```diff
@@ -477,24 +472,12 @@ pub trait MessageRouter {
 	/// Implementations using additional intermediate nodes are responsible for using a
 	/// [`MessageForwardNode`] with `Some` short channel id, if possible. Similarly, implementations
 	/// should call [`BlindedMessagePath::use_compact_introduction_node`].
-	///
-	/// The provided implementation simply delegates to [`MessageRouter::create_blinded_paths`],
-	/// ignoring the short channel ids.
-	fn create_compact_blinded_paths<
+	fn create_blinded_paths<
 		T: secp256k1::Signing + secp256k1::Verification
 	>(
-		&self, recipient: PublicKey, context: MessageContext,
+		&self, recipient: PublicKey, context: MessageContext, blinded_path: BlindedPathType,
```

Comment:
Discussed offline. Probably better to keep this as two separate methods. But still unresolved as to whether how the number of paths used should be specified. Let's wait until that is resolved before reverting though.

Reply:
Yeah, let's revert those two commits and move `BlindedPathType` to `channelmanager`, since it is only needed there.

Reply:
Done!
Reintroduced the two-function for blinded path creation in [pr3246.11](https://github.com/shaavan/rust-lightning/commits/pr3246.11)
Also moved BlindedPathType to channelmanager.
Thanks for the pointer! 🌟